Userland/Libraries/LibJS/Runtime/StringPrototype.cpp

```diff
@@ -559,7 +567,13 @@ JS_DEFINE_NATIVE_FUNCTION(StringPrototype::slice)
 // 22.1.3.21 String.prototype.split ( separator, limit ), https://tc39.es/ecma262/#sec-string.prototype.split
 JS_DEFINE_NATIVE_FUNCTION(StringPrototype::split)
 {
-    // FIXME Implement the @@split part
+    if (auto potential_splitter = vm.argument(0), obj = vm.this_value(global_object);
+        !obj.is_nullish() && !potential_splitter.is_nullish()) {
+        auto splitter = potential_splitter.get_method(global_object, *vm.well_known_symbol_split());
+        if (splitter) {
+            return vm.call(*splitter, potential_splitter, obj, vm.argument(1));
+        }
+    }
```

Comment:
Let's not cram all of this into the `if` condition. Also missing an exception check after GetMethod.

```suggestion
    auto separator = vm.argument(0);
    auto limit = vm.argument(1);
  
    auto* object = require_object_coercible(global_object, vm.this_value(global_object));
    if (vm.exception())
        return {};
  
    if (!separator.is_nullish()) {
        auto splitter = separator.get_method(global_object, *vm.well_known_symbol_split());
        if (vm.exception())
            return {};
        if (splitter)
            return vm.call(*splitter, separator, object, limit);
    }
```

now we can also use `separator` and `limit` instead of `vm.argument()` below.

Reply:
Changed the first if but since limit and separator are already used later in the code with non JS::Value types changed it to separator_argument and limit_argument
Also since we already do require_object_coercible up here made string be non optional below

Reply:
sounds good :)